analytics/property.go

```diff
@@ -0,0 +1,43 @@
+package analytics
+
+import (
+	"github.com/mitchellh/mapstructure"
+)
+
+// Properties ...
+type Properties map[string]interface{}
+
+// Merge ...
+func (p Properties) Merge(properties Properties) Properties {
+	r := Properties{}
+	for key, value := range p {
+		r[key] = value
+	}
+	for key, value := range properties {
+		r[key] = value
+	}
+	return r
+}
+
+// AppendIfNotEmpty ...
+func (p Properties) AppendIfNotEmpty(key string, value string) {
+	if value != "" {
+		p[key] = value
+	}
+}
+
+// ToProperty Converts struct into Properties map based on json tags
+func ToProperty(item interface{}) Properties {
```

Comment:
nitpick: I would propose to rename this function to `newProperty` because from the call site it looks like a constructor and that is how we called them for the ideal step refactor.